src/gt4py/eve/visitors.py

```diff
@@ -121,6 +121,21 @@ def generic_visit(self, node: concepts.RootNode, **kwargs: Any) -> Any:
         return None
 
 
+def _preserve_annex(
+    node: concepts.Node, new_node: concepts.Node, preserved_annex_attrs: tuple[str, ...]
+) -> None:
+    # access to `new_node.annex` implicitly creates the `__node_annex__` attribute in the property getter
+    old_annex = node.annex
+    new_annex_dict = new_node.annex.__dict__
+    for key in preserved_annex_attrs:
+        if (value := getattr(old_annex, key, NOTHING)) is not NOTHING:
+            # Note: The annex value of the new node might not be equal
+            # (in the sense that an equality comparison returns false),
+            # but in the context of the pass, they are equivalent.
+            # Therefore, we don't assert equality here.
+            new_annex_dict[key] = value
```

Comment:
It just occurred to me that the code could be simplified further (or am I missing something?):
```suggestion
    # access to `new_node.annex` implicitly creates the `__node_annex__` attribute in the property getter
    new_annex_dict = new_node.annex.__dict__
    old_annex_dict = node.annex.__dict
    for key in old_annex_dict.keys() & preserved_annex_attrs:
        # Note: The annex value of the new node might not be equal
        # (in the sense that an equality comparison returns false),
        # but in the context of the pass, they are equivalent.
        # Therefore, we don't assert equality here.
        new_annex_dict[key] = old_annex_dict[key]
```
Also after reading your reply  to the other issue below, I don't really understand this comment... Which two values should be semantically equivalent even if not compare true for equality? If you refer to the value in the old and new annex, I don't see how could that happen, given that the value is copied literally here...

Reply:
The value of the annex of the new node and the old node should be equivalent. I omitted the "old node" in the comment to keep the sentence focused, but apparently it became unclear that way. I clarified it. I took your first suggestion, but adopted it such that the new annex value is taken when it already exists. This had been a shortcoming in my original version too, which I overlooked. I've added a test `test_annex_preservation_translated_node_overwritten`.